### ISSUE TITLE
Replace TIS Map

### ISSUE THREAD

**Issue by ghost** [TNRIS/tnris.org#287]
Replace existing TIS map (carto) with new AGOL map. Map has been shared to world and AGOL StratMap Group for editing. Meeting on Tuesday. Someone within AGOL StratMap Group will be maintaining the map.

https://tnris.org/texas-imagery-service/

`<style>.embed-container {position: relative; padding-bottom: 80%; height: 0; max-width: 100%;} .embed-container iframe, .embed-container object, .embed-container iframe{position: absolute; top: 0; left: 0; width: 100%; height: 100%;} small{position: absolute; z-index: 40; bottom: 0; margin-bottom: -15px;}</style><div class="embed-container"><iframe width="500" height="400" frameborder="0" scrolling="no" marginheight="0" marginwidth="0" title="Google Imagery Service" src="//tnris.maps.arcgis.com/apps/Embed/index.html?webmap=410b7caad2ec4e2687b960f488f55ea8&extent=-115.8717,23.9004,-83.4181,38.3197&zoom=true&previewImage=false&scale=true&disable_scroll=true&theme=light"></iframe></div>`

**Comment by ghost** [comment]
I would recommend keeping this pretty simple and adding a field(s) to the county layer data availability and maybe flight date. 

**_Caution:_** this county layer is used in several maps on the site.

**Comment by jwhaney** [comment]
just a quick note here from my experience last time with the map embeds into tnris.org for lidar page, ap, and lp. the embed code you get from ArcGIS Online is not great and requires some tweaking. esri adds some weird classes and other unnecessary junk in there. below are some examples of current ago maps embedded in tnris.org for reference:

__stratmap/elevation-lidar/__
`<iframe width="100%" height="600" frameborder="0" scrolling="no" marginheight="0" marginwidth="0" title="TNRIS.ORG Lidar Coverage Map" src="https://tnris.maps.arcgis.com/apps/Embed/index.html?webmap=66a83e8e93f54068a16477eb4e9f0e75&extent=-108.0433,27.1914,-91.8164,35.3132&home=true&zoom=true&previewImage=false&scale=true&legend=true&disable_scroll=true&theme=light"></iframe>`

__stratmap/address-points/__
`<iframe width="100%" height="400" frameborder="0" scrolling="no" marginheight="0" marginwidth="0" title="Address Points Map" src="https://tnris.maps.arcgis.com/apps/Embed/index.html?webmap=e8c2c6bd86064f6fbadbd38a60b50ccd&extent=-116.3038,22.8609,-83.8502,39.082&home=true&zoom=true&previewImage=false&scale=true&legend=true&disable_scroll=true&theme=light"></iframe>`

__stratmap/land-parcels/__
`<iframe width="100%" height="520" frameborder="0" scrolling="no" marginheight="0" marginwidth="0" title="Land Parcels Map" src="https://tnris.maps.arcgis.com/apps/Embed/index.html?webmap=7f5626b267a042a7869926b3d379f5b9&extent=-116.3038,22.8609,-83.8502,39.082&home=true&zoom=true&previewImage=false&scale=true&legend=true&disable_scroll=true&theme=light"></iframe>`

**Comment by jwhaney** [comment]
> I would recommend keeping this pretty simple and adding a field(s) to the county layer data availability and maybe flight date.
> 
> **_Caution:_** this county layer is used in several maps on the site.

i really like ryan's idea here and it would be very simple to implement and update. the only negative i can think of is that it will not be a true spatial representation of the flight path, but not sure how important that is.

**Comment by ghost** [comment]
Just an update. I'm STILL waiting on the flight blocks shapefile from AppGeo from Hexagon. 2020 Hexagon imagery have not been added to the service yet, but this is anticipated to begin in January starting with the ever popular Panhandle of Texas. I will let @csidenblad know when received so we can switch out the map on the site. Thanks!

**Comment by ghost** [comment]
@csidenblad Good news! The 2020 Hexagon flight block shp is attached here [2020_tx_blocks (2).zip](https://github.com/TNRIS/tnris.org/files/5751430/2020_tx_blocks.2.zip) but I also extracted it here if needed: M:\SM\Imagery\Texas_Imagery_Service\Hexagon\2020_Flight_Blocks\2020_tx_blocks

How about on the map we show **Available    Flight MM/DD/YYYY** (with flight date(s) when we have that) per block and **2020 Coming Soon** for the rest. No 2020 imagery have been put into the service yet.

**Comment by csidenblad** [comment]
@gmullins The shp has been added to a basemap. A new field has been created to show the Available Flight Date. Map can be viewed here.
https://arcg.is/WOunf
Updates can be made to the map after our conversation with Hexagon today.

**Comment by csidenblad** [comment]
Tier 1 and Tier 2 shapes from Hexagon have been added to the folder below: 
M:\SM\Imagery\Texas_Imagery_Service\Hexagon\Tier1and2_AOIs

**Comment by ghost** [comment]
@csidenblad The map looks good. You could go ahead and mark **2020 Available** the 5 blocks on the western edge of the Panhandle since we know those are coming first. And just update with the flight date when we have it. Is there any way to label the flight date **MM/DD/YYYY** directly on the polygons so the user sees that immediately without clicking on it? 

You could also go ahead and load in the 2021 urban shapes we got from Hexagon. It is not important to us to show them broken out as Tier 1 and Tier 2, rather let's show them all as one color/layer. May be easiest to merge them first?
**Urban 2021 Coming Soon** is my suggest label.

Chat me up on this if you like. It is hard to talk cartographic design writing in sentences. Thx!

**IS TEAM:** We will let you know when the map is ready to switch out on the site. Hopefully within the next week.

**Comment by csidenblad** [comment]
https://tnris.maps.arcgis.com/home/webmap/viewer.html?webmap=fd78bc462dfd468f8022ae82a541b7df
TIS map has been updated with both Flight Blocks and Urban Areas with added fields for Available Flights (with requested labels) and Flight Dates (which I have not been able to pull from GIZA preview since AppGeo is currently restructuring the Composite layer to remove 2020 data). Dates will show on blocks when zoomed in, but not a full state level.
@gmullins ready for your review and suggested edits.

**Comment by jwhaney** [comment]
hey @csidenblad, hope you don't mind but i cleaned up some of the popup attributes a bit. i removed the underscores from the title of popups, and changed the sqkm lowercase attribute to 'SqKm' and removed the decimal points from the values because i don't think decimals for such large areas adds valuable information for the user.

I have a couple of suggestions. hopefully that's ok. something for you and @gmullins to think about.
- what value does labelling the flight date provide when it is already in the popup? if you do want the label, i would suggest tweaking the font a bit because it is very thin. if it were me though, i would just have the date in the popup like the rest of the attributes which would keep the map clean. in the future, when there are more dates to use for labels, the map may become cluttered and messy.
- in the urban blocks layer, is there a reason for 'Other'? in the table, i do not see any 'Other' values for the 'Available Flight' attribute/field. if it's not required, i'd say remove/hide it.

these are just my two cents. you can take it however you like, maybe the actual value....two pennies. 😄 

the map looks great!

**Comment by ghost** [comment]
@csidenblad Looking good! I need to give you the Austin/Brazos AOI. It is [here](M:\SM\Imagery\Texas_Imagery_Service\Hexagon\Tier1and2_AOIs\For_map). StratMap21...shp (2 of them). Please merge these in with the urban 2021, same color.

- Speaking of color, what about the 'Urban 2021 Coming Soon' areas as a dark gray, then save the yellow for when we say they are Available. 

@jwhaney Thanks for your suggestions!
- I'm good with the flight date just in the popup. I do see how it could get crowded.
- @csidenblad Perhaps the Other is a place holder for 'Urban 2021 Available'? which I suggest for the yellow.

Still on hold for publication. Thank you!!

**Comment by csidenblad** [comment]
@gmullins The map is updated and ready for review. Changes listed below. 
https://tnris.maps.arcgis.com/home/webmap/viewer.html?webmap=fd78bc462dfd468f8022ae82a541b7df

- The Austin areas have been added. I did merge them as 6in and 12in, separately. All Urban Blocks are now symbolized as dark gray with Available Flight listed as "Urban 2021 Coming Soon".  The yellow will be reserved for 2021 data once received. 
- Flight date has been removed as a label, only exists in pop-up. 
- I have updated the dates for the Hexagon blocks where metadata was available. I do see that these dates are still within the Texas Hexagon Composite 2018 Layer, but the Texas Hexagon 2020 layer is not available within Giza. 

Let me know if you see any other changes that need to be made. :)
